fb_apt/providers/keys.rb

```diff
@@ -0,0 +1,57 @@
+# vim: syntax=ruby:expandtab:shiftwidth=2:softtabstop=2:tabstop=2
+
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+action :run do
+  keyring = node['fb_apt']['keyring']
+  keyserver = node['fb_apt']['keyserver']
+  keys = node['fb_apt']['keys'].to_hash
+
+  if keys && keyring
+    cmd = Mixlib::ShellOut.new("LANG=C apt-key --keyring #{keyring} list")
```

Comment:
Because you won't necessarily have en_US.UTF-8 on a box (you can disable it if you dpkg-reconfigure the locales package or mess with localegen). C is guaranteed to be everywhere (it's a POSIX thing actually) and give you plain output in English, so it's a good choice for parsing.